src/dialogs/stat_dialogs/chart_options_view.pas

```diff
@@ -0,0 +1,275 @@
+unit chart_options_view;
+
+{$mode objfpc}{$H+}
+
+interface
+
+uses
+  Classes, SysUtils, stat_dialog_contribution, StdCtrls, ExtCtrls, EditBtn,
+  Controls, chart_options_model, executor, epidatafilestypes,
+  stat_dialog_custom_view;
+
+type
+
+  { TChartOptionsView }
+
+  TChartOptionsView = class(TCustomStatDialogView)
+  private
+    FDataModel: TChartOptionsModel;
+    FLabel: array of TLabel;
+    FText: array of TCustomEdit;
+    FDate: array of TDateEdit;
+    FOnModified: IStatDiaglogViewModified;
+    FSetXmin,
+    FSetXmax,
+    FSetYmin,
+    FSetYmax:  Boolean;
+  
+    procedure SetText(Sender: TObject);
+    procedure SetDate(Sender: TObject);
+//    procedure UpdateTexts();
+  public
+    constructor Create(TheOwner: TComponent); override;
+    procedure EnterView(); override;
+    function ExitView(): boolean; override;
+    function GetViewCaption(): UTF8String; override;
+    function IsDefined(): boolean; override;
+    procedure ResetView(); override;
+    procedure SetModel(DataModel: TChartOptionsModel);
+    procedure ResetMinMax;
+  end;
+
+implementation
+
+const
+  TITLE_TAG    = Ord(cbT);
+  FOOTNOTE_TAG = Ord(cbF);
+  XTITLE_TAG   = Ord(cbXT);
+  YTITLE_TAG   = Ord(cbYT);
+  COLOR_TAG    = Ord(cbC);
+  XMIN_TAG     = Ord(cbnXMin);
+  XMAX_TAG     = Ord(cbnXMax);
+  YMIN_TAG     = Ord(cbnYMin);
+  YMAX_TAG     = Ord(cbnYMax);
+
+  XDMIN_TAG    = Ord(cbdXMin);
+  XDMAX_TAG    = Ord(cbdXMax);
+  YDMIN_TAG    = Ord(cbdYMin);
+  YDMAX_TAG    = Ord(cbdYMax);
+
+{ TChartOptionsView }
+
+constructor TChartOptionsView.Create(TheOwner: TComponent);
+const
+  // not guaranteed to fit every language, but calculating it from the labels
+  // is not simple (LabelText.Width does not work)
+  labelWidth = 120;
+  dateSize   = 120;
+var
+  EditText:  TCustomEdit;
+  DateText:  TDateEdit;
+  LabelText: TLabel;
+  Labels:    array of UTF8String = ('Main Title', 'Footnote',
+             'X-Axis Title', 'Y-Axis Title', 'Colour Selection',
+             'X-Axis Minumum', 'X-Axis Maximum', 'Y-Axis Minumum', 'Y-Axis Maximum');
+  i:         Integer;
+begin
+  inherited Create(TheOwner);
+
+  SetLength(FText,  Ord(High(TChartOptionEdit)) + 1);
+  SetLength(FDate,  Ord(High(TChartMinMaxDate)) + 1);
+  SetLength(FLabel, Ord(High(TChartOptionEdit)) + 1);
+
+  for i := Low(FText) to High(FText) do
+    begin
+      LabelText := TLabel.Create(TheOwner);
+      LabelText.Caption := Labels[i];
+      LabelText.AutoSize := false;
+      LabelText.Width := labelWidth;
+      LabelText.Alignment := taRightJustify;
+      if (i=0) then
```

Comment:
Unfortunately this still doesn't account for large input fields. The code here makes sure that labels are 10 px. apart, but on Linux the input fields are in general higher and require more spacing.
Ideally it should Edits that anchors to the Edit above, and then the labels that align to the input edit after that.

Reply:
OK. See below. Finally figured out how to position input boxes first, when the min/max boxes can be invisible.